module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,62 @@
+#include "compression.hpp"
+#include <iostream>
+
+void printMap(const Bitmap& bitmap)
+{
+    for (const auto line : bitmap)
+    {
+        for (const auto pixel : line)
+        {
+            std::cout << pixel;
+        }
+        std::cout << "\n";
+    }
+}
+
+CompressedBitmap compressLine(const std::array<uint8_t, width>& line)
+{
+    CompressedBitmap result;
+    result.reserve(width*height);
```

Comment:
Here you need maximum space for only one line, don't you?

Reply:
You compress line, which size is `widht`. So max space in vector you will need is `widht`